libs-scala/resources-akka/src/main/scala/com/digitalasset/resources/akka/AkkaResourceOwnerFactories.scala

```diff
@@ -28,8 +30,23 @@ trait AkkaResourceOwnerFactories[Context] {
   def forCancellable[C <: Cancellable](acquire: () => C): AbstractResourceOwner[Context, C] =
     new CancellableResourceOwner(acquire)
 
-  def forBoundedSourceQueue[T](
-      queueGraph: RunnableGraph[BoundedSourceQueue[T]]
-  )(implicit materializer: Materializer): AbstractResourceOwner[Context, BoundedSourceQueue[T]] =
-    new BoundedSourceQueueResourceOwner[T, Context](queueGraph)
+  def forBoundedSourceQueue[T, U](
+      queueGraph: RunnableGraph[(BoundedSourceQueue[T], Future[U])]
+  )(implicit
+      materializer: Materializer
+  ): AbstractResourceOwner[Context, (BoundedSourceQueue[T], Future[U])] =
+    new BoundedSourceQueueResourceOwner[(BoundedSourceQueue[T], Future[U]), T, Context](
+      queueGraph = queueGraph,
+      toSourceQueue = _._1,
+      toDone = _._2.map(_ => ())(ExecutionContext.parasitic),
+    )
+
+  def forBoundedSourceQueue[T, E](
```

Comment:
Do we need this one? Also we don't invoke it in the tests.

Reply:
Also, why `E` and not something that doesn't look like an error type?

Reply:
Couldn't think of anything.

And I'm not really sure whether we need it or not, because we don't use this code in this repository. So I'm playing it safe.

Reply:
In the worst case we can always monkey-patch it (by putting it under the same package) in the integration kit, which is the only user, I believe. Unless, of course, it makes it to the release.

Reply:
I removed the extra function and renamed the type parameters.

I kept the constructor in case we need to use the "extended" version elsewhere.